src/itdelatrisu/opsu/Container.java

```diff
@@ -93,14 +99,26 @@ public void start() throws SlickException {
 
 		// report any critical errors
 		if (e != null) {
-			ErrorHandler.error(null, e, true);
+			ErrorReport report = new ErrorReport(null, e);
+			ErrorReportCategory info = new ErrorReportCategory("Game Engine Details");
+			info.addSection("State Info", new Callable<String>() {
+				@Override
+				public String call() throws Exception {
+					StateBasedGame sbg = (StateBasedGame)game;
+					return sbg.getCurrentStateID() + " - " + sbg.getCurrentState().getClass().getSimpleName();
+				}
+			});
+			info.addSection("OpenGL software mode", String.valueOf(isSoftwareMode()));
+			report.addInfo(info);
+			ErrorHandler.error(report, true);
+			
 			e = null;
 			forceExit = true;
 		}
 
 		if (forceExit) {
 			Opsu.close();
-			System.exit(0);
+			System.exit(1);
```

Comment:
Forgot to mention in the last review: this isn't necessarily an error (e.g. restarting for an update). You could do:
```
System.exit((e == null) ? 0 : 1);
```